bin/templates/scripts/cordova/lib/run.js

```diff
@@ -0,0 +1,184 @@
+    shell = require('shelljs'),
+    build = require('./build'),
+    spawn = require('./spawn'),
+    check_reqs = require('./check_reqs');
+
+var cordovaPath = path.join(__dirname, '..');
+var projectPath = path.join(__dirname, '..', '..');
+
+module.exports.run = function (argv) {
+
+    // parse args here
+    // --debug and --release args not parsed here
+    // but still valid since they can be passed down to build command 
+    var args  = nopt({
+        // "archs": String,     // TODO: add support for building different archs
+        "list": Boolean,
+        "nobuild": Boolean,
+        "device": Boolean, "emulator": Boolean, "target": String
+    }, {}, argv);
+
+    // Validate args
+    if (args.device && args.emulator) {
+        return Q.reject('Only one of "device"/"emulator" options should be specified');
+    }
+
+    // validate target device for ios-sim
+    // Valid values for "--target" (case sensitive):
+    var validTargets = ["iPhone-4s", "iPhone-5", "iPhone-5s", "iPhone-6-Plus", "iPhone-6",
+        "iPad-2", "iPad-Retina", "iPad-Air", "Resizable-iPhone", "Resizable-iPad"];
+    if (args.target && validTargets.indexOf(args.target) < 0 ) {
+        return Q.reject(args.target + " is not a valid target for emulator");
+    }
+
+    // check for either ios-sim or ios-deploy is available
+    // depending on arguments provided
+    var checkTools = args.device ? check_reqs.check_ios_deploy() : check_reqs.check_ios_sim();
+
+    // support for CB-8168 `cordova/run --list`
+    if (args.list) {
+        if (args.device) return listDevices();
+        if (args.emulator) return listEmulators();
+        return listDevices().then(function () {
+            return listEmulators();
+        });
+    }
+
+    return checkTools.then(function () {
+        if (args.nobuild) {
+            // --nobuild option is passed. Skipping build...
+            return Q();
+        }
+        // if --nobuild isn't specified then build app first
+        return build.run(argv);
+    }).then(function () {
+
+        var xcodeProjFiles = shell.ls(projectPath).filter(function (name) {
+            return path.extname(name) === '.xcodeproj';
+        });
+        
+        if (xcodeProjFiles.length === 0) {
+            return Q.reject("No Xcode project found in " + projectPath);
+        }
+        if (xcodeProjFiles.length > 1) {
+            console.warn('Found multiple .xcodeproj directories in \n' +
+                projectPath + '\nUsing first one');
+        }
+
+        var projectName = path.basename(xcodeProjFiles[0], '.xcodeproj');
+
+        var appPath = path.join(projectPath, 'build', (args.device ? 'device' : 'emulator'), projectName + '.app');
+        // select command to run and arguments depending whether
+        // we're running on device/emulator
+        if (args.device) {
+            return checkDeviceConnected().then(function () {
+                return deployToDevice(appPath);
+            }).fail(function () {
```

Comment:
This will deploy to simulator if deployToDevice() fails. I think we want it to only do this if checkDeviceConnected() fails. So, you should:

    return checkDeviceConnected().then(function () {
        return deployToDevice(appPath);
    }, function () {
        return deployToSim(appPath, args.target);
    });